assets/collage.css

```diff
@@ -154,9 +148,35 @@
   border: 0;
 }
 
-.collage-card .deferred-media__poster:focus,
-.collage-card .collage-card__link:focus {
+.collage-card .deferred-media__poster:after {
+  content: "";
+  position: absolute;
+  z-index: 1;
+  bottom: calc(var(--card-border-width) * -1);
+  left: calc(var(--card-border-width) * -1);
+  right: calc(var(--card-border-width) * -1);
+  top: calc(var(--card-border-width) * -1);
   outline-offset: 0.3rem;
+}
+
+.collage-card .deferred-media__poster:focus:after {
+  box-shadow: 0 0 0 0.3rem rgb(var(--color-background)),0 0 0.5rem 0.4rem rgba(var(--color-foreground), 0.3);
+  outline: 0.2rem solid rgba(var(--color-foreground), 0.5);
+}
+
+.collage-card .deferred-media__poster:focus-visible:after {
+  box-shadow: 0 0 0 0.3rem rgb(var(--color-background)),0 0 0.5rem 0.4rem rgba(var(--color-foreground), 0.3);
+  outline: 0.2rem solid rgba(var(--color-foreground), 0.5);
+}
+
+.collage-card .deferred-media__poster:focus:not(:focus-visible),
+.collage-card .deferred-media__poster:focus:not(:focus-visible):after,
+.collage-card .collage-card__link:focus:not(:focus-visible) {
```

Comment:
I don't think you don't need this class here.
This class is only used if there is no-js and we have styling in our global css file that should handle this: 

```
.no-js :focus:not(:focus-visible) {
    outline: 0;
    box-shadow: none;
}
```